nnfabrik/templates/transfer/transferred_trained_model.py

```diff
@@ -0,0 +1,310 @@
+import tempfile
+import torch
+import os
+import datajoint as dj
+import numpy as np
+from nnfabrik.templates.checkpoint import TrainedModelChkptBase
+from nnfabrik.utility.dj_helpers import gitlog, make_hash
+from nnfabrik.templates.trained_model import TrainedModelBase
+
+
+class TransferredTrainedModelBase(TrainedModelBase):
+    """
+    A modified version of TrainedModel table which enables step-wise table population given the
+    step specification. Refer to the corresponding example notebook for a demo.
+    """
+
+    table_comment = "Transferred trained models"
+
+    @property
+    def definition(self):
+        definition = """
+        # {table_comment}
+        transfer_step:                     int          # transfer step
+        -> self().model_table
+        -> self().dataset_table
+        -> self().trainer_table
+        -> self().seed_table
+        collapsed_history:                 varchar(64)
+        data_transfer:                     tinyint 
+        ---
+        comment='':                        varchar(768) # short description 
+        score:                             float        # loss
+        output:                            longblob     # trainer object's output
+        ->[nullable] self().user_table
+        trainedmodel_ts=CURRENT_TIMESTAMP: timestamp    # UTZ timestamp at time of insertion
+        """.format(
+            table_comment=self.table_comment
+        )
+        return definition
+
+    class ModelStorage(TrainedModelBase.ModelStorage):
+        pass
+
+    class DataStorage(dj.Part):
+        @property
+        def definition(self):
+            definition = """
+            # Contains the data generated by the transfer step, stored externally.
+            -> master
+            ---
+            transfer_data:            attach@{storage}
+            """.format(
+                storage=self._master.storage
+            )
+            return definition
+
+    class CollapsedHistory(dj.Part):
+        """
+        For the result of two or more transfer steps to be uniquely identifiable,
+        we compress its entire history (the keys of all previous steps) into a single hash (`collapsed_history`).
+        This table keeps track of this process and can be used to recursively retrieve the transfer history.
+        """
+
+        definition = """
```

Comment:
a recommendation is to make this a proper part table by referring to `-> master`. In doing so, you would need to have a new attribute (e.g. `next_collapsed_history`) to house the new value of `collapsed_history` that reflects the model that corresponds to the `-> master`. (And with that you don't need `prev_collapsed_history`)